### PR TITLE
Refactor Primitives to use Count String

### LINKED ISSUE #181: Refactor primitives to use `count_string` where applicable
Some of the primitives in `nlp_primitives` can be rewritten to use the `CountString` primitive's functionality. This would reduce code duplication and could potentially help performance. A few ones to investigate are: 

- `TitleWordCount`
- `UpperCaseCount`
- `PunctuationCount`

### PR DESCRIPTION
**Pull request by sbadithe**
- Fixes #181 
- Refactors `TitleWordCount`, `PunctuationCount`, `UpperCaseCount` to use `count_string`

### CODE REVIEW

File: docs/source/changelog.rst [hunk 1]
```diff
@@ -5,6 +5,7 @@ Changelog
 Future Release
 ==============
     * Enhancements
+        * Change `TitleWordCount`, `PunctuationCount`, `UpperCaseCount` to use `CountString` (:pr:`183`)
```
Comment (thehomebrewnerd) [inline docs/source/changelog.rst:8]: I'd move this to the `Changes` section. Enhancements is generally reserved for more user-related enhancements, not code improvements.

**Review by gsheni — DISMISSED** [review]
lgtm

File: nlp_primitives/punctuation_count.py [hunk 2]
```diff
@@ -32,14 +32,6 @@ class PunctuationCount(TransformPrimitive):
     return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
```
Comment (thehomebrewnerd) [inline nlp_primitives/punctuation_count.py:32]: Can we use `IntegerNullable` here instead of `Double`?

File: nlp_primitives/title_word_count.py [hunk 3]
```diff
@@ -28,14 +28,6 @@ class TitleWordCount(TransformPrimitive):
     return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
```
Comment (thehomebrewnerd) [inline nlp_primitives/title_word_count.py:28]: Can we use `IntegerNullable` here instead of `Double`?

File: nlp_primitives/upper_case_count.py [hunk 4]
```diff
@@ -27,14 +27,6 @@ class UpperCaseCount(TransformPrimitive):
     return_type = ColumnSchema(logical_type=Double, semantic_tags={"numeric"})
```
Comment (thehomebrewnerd) [inline nlp_primitives/upper_case_count.py:27]: `Double` -> `IntegerNullable`?

**Review by thehomebrewnerd — APPROVED** [review]
LGTM

**Pull request merged**